Documentation outline

perrygeo opened the pull request:
This is far from being the complete documentation - only proposing a basic structure and layout of the sphinx docs here.

A preview available here: https://mapbox.s3.amazonaws.com/playground/perrygeo/rasterio-docs/index.html#

:eyes: needed ... specifically feedback on the general structure/outline.

Resolves #488 
Resolves #584 
Resolves #564

Related to #37, #390, #451

brendan-ward commented on docs/community.rst:
```diff
@@ -0,0 +1,43 @@
+Code of conduct
+---------------
+
+.. include:: ../CODE_OF_CONDUCT.md
```
@perrygeo this brings in a second, escaped header that reads as 
`# Contributor Code of Conduct`

And doesn't render the link at the end properly:
...` [Contributor Covenant](http://contributor-covenant.org), version 1.2.0, available at [http://contributor-covenant.org/version/1/2/0/](http://contributor-covenant.org/version/1/2/0/)`

Perhaps converting this from a markdown to a reStructuredText file would allow proper display via Github and clean inclusion here?

brendan-ward replied:
@perrygeo  what about putting Code of Conduct first under the community section here and in TOC?  It is the one subsection not quite like the others, so first or last seems best.

perrygeo replied:
Starting with the code of conduct seems a little in-your-face ... but it is awkward in the middle so I'll scoot it to the end.

brendan-ward commented on docs/community.rst:
```diff
@@ -0,0 +1,43 @@
+
+* `Mapbox`_ uses rasterio to process satellite imagery for it's global satellite base maps.
+* Mary Marek-Spartz's masters thesis compares the map algebra tools of the proprietary ESRI ArcPy library and the open source Rasterio library.  https://github.com/alfalimajuliett/masters-thesis
+
```
@perrygeo might be nice to include a brief statement on how to get your project or use case listed here.

Please add:
* The Conservation Biology Institute (http://consbio.org/) uses rasterio to provide geoprocessing of NetCDF files (https://github.com/consbio/clover).

brendan-ward commented on docs/community.rst:
```diff
@@ -0,0 +1,43 @@
+Who are the rasterio developers?
+--------------------------------
+
+.. include:: ../AUTHORS.txt
```
@perrygeo this doesn't render particularly well either.  Another candidate to convert to reStructuredText format?

brendan-ward commented on docs/conf.py:
```diff
@@ -0,0 +1,367 @@
+# built documents.
+#
+# The short X.Y version.
+version = u'1.0'
```
@perrygeo should this be 0.32.0 instead?

perrygeo replied:
+1 I'll set this automatically ala https://github.com/mapbox/rasterio/blob/master/setup.py#L53  -

brendan-ward commented on docs/cookbook.rst:
```diff
@@ -0,0 +1,3 @@
+=================
+Rasterio Cookbook
```
@perrygeo should this have more to at, at least a `.. todo::` and some brief ideas, or should this file be left out at this point if it is empty?

jdmcbr replied:
@perrygeo Is there an ideal format for cookbook entries? And, is it preferable if it relies only on `rasterio`, or would something using `fiona` and `shapely` as well be acceptable? The first application I had for all of these libraries (and by far my most common use case still) is masking rasters using shapefiles. I pieced together code @sgillies had posted in various places. I'd be happy to put it together into a recipe for that specific task.

perrygeo replied:
@jdmcbr `.rst` format would be best . In terms of the content, I was thinking bite sized pieces of code (like ~40 lines) for doing discrete chunks of work with rasterio. Not complete workflows but little snippets that are generally useful templates for common tasks. Showing integration with other packages is great but the focus should be rasterio usage. 

I'm picturing something along the lines fof https://pcjericks.github.io/py-gdalogr-cookbook/raster_layers.html - but without  osgeo.gdal

brendan-ward commented on docs/index.rst:
```diff
@@ -0,0 +1,57 @@
+.. rasterio documentation master file, created by
```
@perrygeo are the leading comments necessary or can they be removed?

perrygeo replied:
just sphinx cruft, removed

brendan-ward commented on docs/intro.rst:
```diff
@@ -0,0 +1,4 @@
+Introduction
```
@perrygeo also a `.. todo::`?  I didn't see this linked from any of the other pages here; perhaps I missed something?

perrygeo replied:
Oh yeah, missed that one. The intention was for the intro to go in `index` before the table of contents. I'll remove this.

brendan-ward commented on docs/topics.rst:
```diff
@@ -0,0 +1,23 @@
+Topics
```
@perrygeo how about "General Concepts and Operations"?

brendan-ward commented on docs/developers.rst:
```diff
@@ -0,0 +1,7 @@
+
+.. todo::
+
+    Everything developers need to know to get a dev environment running,
```
@perrygeo for now, this could crosslink to some of the wiki pages: 

* https://github.com/mapbox/rasterio/wiki/Development-Guide
* https://github.com/mapbox/rasterio/wiki/Exposing-GDAL-Functionality
* https://github.com/mapbox/rasterio/wiki/Cython-and-GDAL

Though a longer term goal of consolidating that information into the docs is a good idea.

brendan-ward commented on docs/python_manual.rst:
```diff
@@ -0,0 +1,26 @@
+    working_with_datasets
+    writing
+    topics
+    osgeo_gdal_migration
```
@perrygeo this seems like it should go to the end of the list (here and in the main TOC)

brendan-ward commented:
@perrygeo thank you for doing this, this is a big step toward 1.0.  

Are you wanting the TODO items sorted out in this PR, or are you planning to create new issues and deal with them later?  If we are OK with waiting, some of these might be perfect things to bite off for the upcoming sprint at PyCon...

brendan-ward commented on docs/index.rst:
```diff
@@ -0,0 +1,57 @@
+Its functions typically accept and return Numpy ndarrays.
+Rasterio is designed to make working with geospatial raster data more productive and more fun.
+
+You can download from pypi and todo quick link to install
```
@perrygeo is this TODO intentional here, or can it be linked to install page now?

perrygeo commented:
Thanks @brendan-ward! I think I addressed most of your comments in the latest commit.

> Are you wanting the TODO items sorted out in this PR, or are you planning to create new issues and deal with them later? If we are OK with waiting, some of these might be perfect things to bite off for the upcoming sprint at PyCon...

Yeah, I'm OK with waiting. My hope is that having the structure will encourage all of us (and new contributors/sprint attendees) to bite off reasonable chunks, tackle the docs one piece at a time. No one wants a multi-week documentation death march!

brendan-ward commented:
@perrygeo this looks good, thanks for updating.
